src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -213,6 +204,37 @@ function createEmscriptenModuleInstance(loadAssemblyUrls: string[], onReady: ()
   });
 
   return module;
+
+  async function addResourceAsAssembly(dependency: LoadingResource) {
+    const runDependencyId = `blazor:${dependency.name}`;
+    Module.addRunDependency(runDependencyId);
+
+    try {
+      // Wait for the data to be loaded and verified
+      const dataBuffer = await dependency.data;
+
+      // Load it into the Mono runtime
+      const data = new Uint8Array(dataBuffer);
+      const heapAddress = Module._malloc(data.length);
+      const heapMemory = new Uint8Array(Module.HEAPU8.buffer, heapAddress, data.length);
+      heapMemory.set(data);
+      mono_wasm_add_assembly(dependency.name, heapAddress, data.length);
```

Comment:
Are we able to use the mono APIs to load assemblies: https://github.com/mono/mono/blob/4cb6ca559996aa31bfdd8d9a5658761e097df9bd/sdks/wasm/src/library_mono.js#L161? It looks like you can use the fetch callback to read from the `resourceLoader`

Reply:
I prefer our logic. The mono one logs a lot of unnecessary stuff.